### ISSUE TITLE
Semantic label in Material DateTimePicker Icons

### ISSUE THREAD

**Issue by burhanrashid52** [flutter/flutter#153562]
### Use case

I am currently writing end-to-end tests using [Maestro](https://maestro.mobile.dev/). Maestro interacts with a Flutter app either [using text or semantic labels](https://maestro.mobile.dev/platform-support/flutter#interacting-with-widgets-by-semantic-identifier) for `Icons` and `Buttons`. 

Currently, I am writing a test which involves a date picker that I want to control in Maestro to select a specific date. This can be simply done by using `semanticLabels` on `Icons` used in `DateTimePicker` but unfortunately, it is not currently available.

### Proposal

Add meaningful semantics label to this icons

back icon => `previous_month`
forward icon => `next_month`
down_arrow => `switch_year`
calendar icon => `switch_input_date`
edit icon => `switch_calendar_month`

And so on. If this look good I can create a PR for it.

**Comment by Piinks** [comment]
We think this might be good for the new accessibility triage (🙂 )
Feel free to direct back if not! Thanks!

**Comment by burhanrashid52** [comment]
@Piinks Thanks...Should I create a PR or ask someone for more info ?

**Comment by chunhtai** [comment]
cc @hangyujin can you verify if this is the case? we also have a datepicker use case in a11y_assessment but for some reason it didn't pick up this issue.

**Comment by YungCatz** [comment]
Is there any new Information or Development Process for this Issue?
I think it is a pretty big Issue, because of the lack of accessibility.

**Comment by Piinks** [comment]
Per our wiki on [issue hygiene](https://github.com/flutter/flutter/tree/master/docs/contributing/issue_hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-issues-or-prs):

> Do not add "me too" or "same" or "is there an update" comments to issues or PRs

Continued posts like this are not constructive and will result in restrictions to make further comments in this repository.

**Comment by ChristianEdwardPadilla** [comment]
I believe this issue would need to be fixed in order to fix the following related bad a11y behavior (b/382542467):

**Expected result:**
The "Previous month" and "Next month" buttons should be announced with an informative label.

**Actual result:**
The "Previous month" and "Next month" buttons are announced as "Button".

<details>
  <summary>Repro project</summary>
  
```
import 'dart:async';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  DateTime currentDate = DateTime.now();

  Future<void> _selectDate(BuildContext context) async {
    final pickedDate = await showDatePicker(
      context: context,
      initialDate: currentDate,
      firstDate: DateTime(2015),
      lastDate: DateTime(2050),
    );
    if (pickedDate != null && pickedDate != currentDate) {
      setState(() {
        currentDate = pickedDate;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("DatePicker"),
      ),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(currentDate.toString()),
            ElevatedButton(
              onPressed: () => _selectDate(context),
              child: const Text('Select date'),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>


<details>
  <summary>Images showing "Button" announcement</summary>

| | |
| --- | --- |
| ![1B](https://github.com/user-attachments/assets/ac1fb8d3-3c0f-4c65-b15e-41404b0a23d0) | ![1A](https://github.com/user-attachments/assets/934e1b78-57b1-4ba1-a568-af301918b3f4) |

</details>

**Comment by burhanrashid52** [comment]
@ChristianEdwardPadilla Since this issue is assigned to a Flutter Team memeber, does this mean they will fix the issue and raise a PR? Or can I do it myself?

**Comment by hannah-hyj** [comment]
@burhanrashid52 I haven't started working on this issue, feel free to file a PR and take it over.

**Comment by hannah-hyj** [comment]
@burhanrashid52 I now started working on this issue

**Comment by hannah-hyj** [comment]
looks like these icon buttons are using `tooltip` instead of `label` to announce their functions. 
The tree looks like this 
The edit icon:
   │     │   └─SemanticsNode#14
   │     │       Rect.fromLTRB(0.0, 0.0, 48.0, 48.0)
   │     │       actions: focus, tap
   │     │       flags: isButton, hasEnabledState, isEnabled, isFocusable
   │     │       tooltip: "Switch to input"
   │     │       textDirection: ltr

The previous month and next month icons: 

   │     │ ├─SemanticsNode#16
   │     │ │   Rect.fromLTRB(228.0, 2.0, 276.0, 50.0)
   │     │ │   actions: focus, tap
   │     │ │   flags: isButton, hasEnabledState, isEnabled, isFocusable
   │     │ │   tooltip: "Previous month"
   │     │ │   textDirection: ltr
   │     │ │
   │     │ ├─SemanticsNode#17
   │     │ │   Rect.fromLTRB(276.0, 2.0, 324.0, 50.0)
   │     │ │   actions: focus, tap
   │     │ │   flags: isButton, hasEnabledState, isEnabled, isFocusable
   │     │ │   tooltip: "Next month"
   │     │ │   textDirection: ltr

I tested on android talk back and iOS voice over, it works well, focus on the icon button will first announce the label and role, then announce the tooltip. 

@burhanrashid52 If it's just an issue when Maestro interacts with a Flutter app, maybe you can update the Maestro function to also include semantics tooltips, but I will close this as this doesn't seem like a flutter issue.
